src/codegen/mod.rs

```diff
@@ -98,6 +98,12 @@ fn root_import(
 struct CodegenResult<'a> {
     items: Vec<proc_macro2::TokenStream>,
 
+    /// Just the layout tests.
+    ///
+    /// Used to implement `LayoutTests::EmitOnly`, and empty if that setting is
+    /// disabled (note that conversely `items` is always populted).
```

Comment:
Note to self: fix typo here.

It's a little unfortunate that `items` is written even in cases where we only want the test, but I think it would be a way bigger patch otherwise, since there are so many places to change. This didn't feel like the right tradeoff for this feature.

Maybe I'm wrong though. But... I'd argue it seems like the kind of thing to file as a potential optimization someone can do if they care or something. (E.g. please don't ask me to do this here unless it's very necessary 😅)